grid2op/tests/test_ma.py

```diff
@@ -651,278 +655,245 @@ def check_action_spaces(self, ma_env):
                 ==\
                     ma_env._subgrids_cls['action'][agent].interco_pos_topo_vect).all() 
         
-    def are_same_actions(self, a1, a2, type_action, add_msg = ""):
-        # TODO COMMENTS before each
+    def are_same_actions(self, a1, a2, type_action, 
+                         add_msg = ""):
+        # Checks both actions have the same action type
+        # meaning if the same 
+        assert (getattr(a1, type_action) == getattr(a2, type_action)).all(), add_msg
+        # Chacks if the same action type is applied 
+        assert getattr(a1, f'_modif_{type_action}') == getattr(a2, f'_modif_{type_action}'), add_msg
+        # Checks if the actions are the same
+        # meaning they have the same effect
+        # on the grid
         assert a1 == a2, add_msg
-        assert eval(f'(a1.{type_action} == a2.{type_action}).all()'), add_msg
-        assert eval(f'a1._modif_{type_action} == a2._modif_{type_action}'), add_msg
+        
+        
+    def check_local2global(self, agent, n, orig_ids, type_action, is_set, value,
+                           assert_type = None, assert_modif = None):
+        # We check for every local id if _local_action_to_global 
+        # gives the correspondig global action on the same object
+        # and they have the same effect.
+        
+        for local_id in range(n):
```

Comment:
what's going on if n is an index superior to the maximum local id

Reply:
explicit is better than implicit.

So either write a comment about that or assert the right things, thanks

Reply:
ok for now. For later, since you pass agent, you could also retrieve n and orig_ids with self.ma_env._subgrids_cls['action'][agent] by specifying the asset you are doing action on.

Reply:
I just added this assert : 
`assert n == len(orig_ids)`